### PR TITLE
fix: validate plugin configuration in the DP

### LINKED ISSUE #2834: bug: missing JSONSchema validation for plugin attributes
### Issue description

Currently, only Route/Service self is checked. The bound Plugins were not be checked now.
That is wrong.

https://github.com/apache/apisix/blob/master/apisix/http/router/radixtree_uri.lua#L118

In this way, when the data plane is running the plug-in, it exits abnormally due to illegal plug-in parameters.

1. use JSON schema to check the plugin, confirm the input parameters are valid.
2. set the default value of the plug-in through the JSON schema check.

### PR DESCRIPTION
**Pull request by spacewander**
Fix #2834

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

**Comment by membphis** [comment]
wow,  That is a nice PR

**Comment by membphis** [comment]
@spacewander  Please update the checklist

**Comment by membphis** [comment]
@spacewander you can continue this issue

**Review by membphis — APPROVED** [review]

File: apisix/core/config_etcd.lua [hunk 1]
```diff
@@ -256,6 +264,14 @@ local function sync_data(self)
                     end
                 end
 
+                if data_valid and self.checker then
+                    data_valid, err = self.checker(item.value)
+                    if not data_valid then
+                        log.error("failed to check item data of [", self.key,
+                                "] err:", err, " ,val: ", json.delay_encode(item.value))
```
Comment (tokers) [inline apisix/core/config_etcd.lua:271]: Bad indentation.

File: apisix/plugin.lua [hunk 2]
```diff
@@ -498,4 +498,102 @@ function _M.get(name)
 end
 
 
+local function check_schema(plugins_conf, schema_type, skip_disabled_plugin)
+    for name, plugin_conf in pairs(plugins_conf) do
+        core.log.info("check plugin scheme, name: ", name, ", configurations: ",
```
Comment (tokers) [inline apisix/plugin.lua:503]: Typo: `scheme` => `schema`.

File: apisix/plugin.lua [hunk 3]
```diff
@@ -498,4 +498,102 @@ function _M.get(name)
...
+    end
+
+    return true
+end
+_M.check_schema = check_schema
+
+
+local function stream_check_schema(plugins_conf, schema_type, skip_disabled_plugin)
+    for name, plugin_conf in pairs(plugins_conf) do
+        core.log.info("check stream plugin scheme, name: ", name,
```
Comment (tokers) [inline apisix/plugin.lua:543]: Ditto.

File: apisix/core/config_etcd.lua (hunk 1 above)
Comment (spacewander) [reply apisix/core/config_etcd.lua:271]: Updated.

**Review by moonming — APPROVED** [review]

**Pull request merged**